### PR TITLE
Adjust Anchor to use transient props

### PR DESCRIPTION
**Pull request by taysea**
#### What does this PR do?

Adjusted Anchor to use transient props. Snapshot changes are expected based on styled components v6 removing vendor prefixes.

#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `userEvent` is used in place of `fireEvent`.
- [ ] `asFragment()` is used for snapshot testing.

### CODE REVIEW

File: src/js/components/Anchor/__tests__/__snapshots__/Anchor-test.tsx.snap [hunk 1]
```diff
@@ -1444,26 +1319,24 @@ exports[`Anchor renders tag 1`] = `
...
-  -webkit-text-decoration: underline;
   text-decoration: underline;
 }
 
 <div>
   <div
     class="c0"
   >
-    <span
+    <a
```
Comment (jcfilben) [inline src/js/components/Anchor/__tests__/__snapshots__/Anchor-test.tsx.snap:1334]: It looks like the `as='span'` prop in this test is being ignored

Comment (taysea) [reply src/js/components/Anchor/__tests__/__snapshots__/Anchor-test.tsx.snap:1334]: good catch

Comment (taysea) [reply src/js/components/Anchor/__tests__/__snapshots__/Anchor-test.tsx.snap]: Fixed, needed to destructure as

Comment (jcfilben) [reply src/js/components/Anchor/__tests__/__snapshots__/Anchor-test.tsx.snap]: Can we make a note in our styled components google doc that for components that support `as` we need to make sure they are destructured

File: src/js/components/Anchor/Anchor.js [hunk 2]
```diff
@@ -18,10 +18,12 @@ import { StyledAnchor } from './StyledAnchor';
...
 import { useAnalytics } from '../../contexts/AnalyticsContext';
 import { TextContext } from '../Text/TextContext';
+import { convertRestToTransientProps } from '../../utils/styles';
 
 const Anchor = forwardRef(
   (
     {
+      as,
       a11yTitle,
       'aria-label': ariaLabel,
```
Comment (jcfilben) [inline src/js/components/Anchor/Anchor.js:28]:
```suggestion
      a11yTitle,
      'aria-label': ariaLabel,
      as,
```

File: src/js/components/Anchor/Anchor.js [hunk 3]
```diff
@@ -87,14 +90,12 @@ const Anchor = forwardRef(
     return (
       <StyledAnchor
         {...rest}
+        as={as}
         ref={ref}
         aria-label={ariaLabel || a11yTitle}
-        colorProp={color}
+        // disabled is not supported by native HTML,
+        // but keeping because is passes @emotion/is-prop-valid check
         disabled={disabled}
```
Comment (jcfilben) [inline src/js/components/Anchor/Anchor.js:98]: Can we alphabetize?

Comment (taysea) [reply src/js/components/Anchor/Anchor.js]: Done.

File: src/js/components/Anchor/__tests__/__snapshots__/Anchor-test.tsx.snap (hunk 1 above)
Comment (taysea) [reply src/js/components/Anchor/__tests__/__snapshots__/Anchor-test.tsx.snap]: Done.

**Review by jcfilben — APPROVED** [review]

**Pull request merged**